src/utils/errors.ts

```diff
@@ -36,16 +36,38 @@ const requestQuery = (response: HttpResponse): string => {
 }
 
 export const extractErrorMessageFromEnvelope = (z: ZObject, response: HttpResponse): ErrorDetails => {
-  const errorResponse = parseErrorResponse(z, response.content)
   return {
     statusCode: response.status,
-    statusMessage: extractStatusMessage(errorResponse),
-    errorDetails: extractErrorDetails(errorResponse),
+    ...buildErrorMessageDetails(z, response),
     query: requestQuery(response),
     url: response.request.url
   }
 }
 
+export const buildErrorMessageDetails = (z: ZObject, response: HttpResponse): { statusMessage: string, errorDetails: string[] } => {
+  const {status} = response
+  if (isRateLimitReached(status)) {
+    return {
+      statusMessage: '429 Too Many Requests',
+      errorDetails: ['You reached rate limit for Sell API']
+    }
+  }
+  // Purpose of this layer is to not break Zapier App in case of returning malformed envelope from Public API
+  try {
+    const errorResponse = parseErrorResponse(z, response.content)
+    return {
+      statusMessage: extractStatusMessage(errorResponse),
```

Comment:
We're handling all statuses `extractStatusMessage` and `extractErrorDetails` but we need to add separate if for `isRateLimitReached`? I'm wondering if this should be handled in one place

Reply:
Yep, you are right. Will move all those to single place